### PR TITLE
Add mono features to fix for new game version

### PR DESCRIPTION
**Pull request by ero-qt**

### CODE REVIEW

File: heave_ho.asl [hunk 1]
```diff
@@ -2,178 +2,155 @@ state("HeaveHo") {}
...
-    }
-
-    print("[ASL] GameManager::Awake found at: 0x" + address.ToString("X16"));
-
-    // Get game manager instance
-    IntPtr mGameManager = new IntPtr(memory.ReadValue<long>(address + gmOffset));
-    vars.GameManager = memory.ReadPointer(mGameManager);
-    print("[ASL] GameManager::Instance found at: 0x" + vars.GameManager.ToString("X16"));
-}
+	var classes = new Dictionary<string, uint>
```
Comment (TheTedder) [inline heave_ho.asl:24]: What's even the point of a dictionary with a single member? Just do 
`uint gameManagerToken = 0x200001D;`
or something if you really want to.

File: heave_ho.asl [hunk 2]
```diff
@@ -2,178 +2,155 @@ state("HeaveHo") {}
...
+					var vtable_size = game.ReadValue<int>(klass + 0x5C);
+					mono.Add(token.Key, new DeepPointer(klass + 0xD0, 0x8, 0x40 + 0x8 * vtable_size).Deref<IntPtr>(game));
+
+					vars.Log("Found " + token.Key + " at 0x" + mono[token.Key].ToString("X") + ".");
+				}
+			}
+
+			if (mono.Count == classes.Count && mono.Values.All(ptr => ptr != IntPtr.Zero))
+			{
+				vars.Data = new MemoryWatcherList
```
Comment (TheTedder) [inline heave_ho.asl:88]: Don't need a whole list of memory watchers if we only have one.

**Review by TheTedder — CHANGES_REQUESTED** [review]
some stuff to clean up. seems like you put some unnecessary boilerplate code in. Also, try to avoid creating single-use DeepPointers. Instead, use `ReadPointer()`. You also dereference the same offsets multiple times in some places.

**Comment by ero-qt** [comment]
cant be bothered

**Pull request closed without merging**